### DISCUSSION TITLE
(Newbie) Having trouble with setup

### DISCUSSION THREAD

**Discussion by Kry0hh** [LiveSplit/LiveSplitOne#610] [Help]
Fairly recently I've made the switch over from windows. Usually I'm able to troubleshoot on my own with this sort of thing, but I've never had to build like this before. I have Rust Compiler and NPM all setup, but I'm having trouble when trying to run npm start. This is the output I get:
![Screenshot_20210116_093651](https://user-images.githubusercontent.com/76639652/104828782-cb207880-5821-11eb-988f-5514220f7b53.png)

Up until this point, everything went alright. No errors or anything that weren't just stupid little mistakes on my end. Hoping this is just another stupid mistake haha
Sorry for such a newbie question. But any help would be greatly appreciated! 

Also, if it helps. I'm on Manjaro, so I'm Arch-based, and here's the log file from the above screenshot:
[log_file.log](https://github.com/LiveSplit/LiveSplitOne/files/5825417/log_file.log)

**Comment by CryZe** [comment]
Can you try running `/bin/sh -c "cargo run"` manually? It seems like that command fails for whatever reason. Either /bin/sh doesn't work, or somehow it can't find cargo or something.

Also as a side note: Is there a specific reason you want to build it yourself? If you just want to use it, there shouldn't be much reason to build it yourself.

**Comment by Kry0hh** [comment]
Honestly, just assumed I needed to build it myself to use it properly. I knew that there's a web version, but I can't use hotkeys while tabbed in-game that way, so I figured I'd try building it myself just in case. Like I said before, I'm fairly new to Linux. Windows raised me ignorant to a lot of things, and resources on this particular issue were kind of hard to find haha.
Anyways, running that command tells me: 
error: could not find `Cargo.toml` in `/usr/bin` or any parent directory

**Comment by wooferzfg** [comment]
@Kry0hh Can you try running that command in the `livesplit-core` directory?

**Comment by CryZe** [comment]
oh wait, is the submodule maybe just not checked out at all? Not sure if that's what the error would look like, but it's certainly a possible source of problems.

**Comment by Kry0hh** [comment]
@wooferzfg I gave it a shot, sadly same error.. although, I will say that my livesplit-core folder is completely empty which doesn't seem right? I know 0 JS, so I could be wrong, but when I looked it up it it said cwd is to change the working directory. It's trying to to that to livesplit-core/capi/bind_gen, but there's no other files or folders inside livesplit-core right now.

**Comment by Kry0hh** [comment]
@CryZe @wooferzfg Also sorry for my slow responses, but thanks for both your help. How would I check for that @CryZe ?

**Comment by DarkRTA** [comment]
> I will say that my livesplit-core folder is completely empty which doesn't seem right? 

You do not have the submodules checked out. Try running the following then building again.
```
git submodule init
git submodule update
```

I normally clone repos with `git clone --recursive` in order for this to be done automatically.

**Comment by Kry0hh** [comment]
@DarkRTA I literally don't know HOW in the world I didn't clone the repo with recursive, since I'd done every command over a couple times while troubleshooting just to make sure... but yeah, that seems to have worked. I executed npm run build:core successfully! 
New issue though, it doesn't open a web browser when I npm run start. Although, it's not giving any error or anything. It just stops at 
ℹ ｢wdm｣: Compiled successfully.
Could I manually grab the websocket url and put it into the livesplit one site or something?

**Comment by mgolisch** [comment]
it probably failed to figure out how to run your browser.

the first few lines of output  should contain the url the development server is listening on. just open that in your browser.

But as @CryZe already said theres no difference than using the hosted version. it is exactly the same thing.
You will still have the problem that the hotkeys/keyboard events will only work if the browser tab is focused.

The only advantage of hosting livesplitone yourself would be either you want to hack on it or maybe that when you write a websocket server to control it it would not have to be using secure websocket protocol.(#495 )

So in either way youd have to write a websocket server that livesplitone can connect to and that sends messages to control it when you press your hotkeys(Or search if someone already did this)

there is a desktop version based on livesplit-core in the works but i dont think it is ready for general use yet.

**Comment by mgolisch** [comment]
out of curiousity hacked togheter a simple example

https://github.com/mgolisch/lso-remote

1) pip install --user pipx
2) pipx install git+https://github.com/mgolisch/lso-remote
3) lso-remote run-server
4) connect local lso or one.livesplit.org to ws://localhost:5000 (seems to work fine in chromium atleast)
5) send commands using lso-remote send-command <name> (split/skip/reset .. see --help)
6) use hotkey app of choice (DE/WM default or something like https://github.com/baskerville/sxhkd) to bind keys to the corresponding command
7) profit

**Comment by thecodewarrior** [comment]
Yeah, it seems both Firefox and Chrome have exceptions for SSL when talking to localhost. https://github.com/LiveSplit/LiveSplitOne/issues/275#issuecomment-770020039
